ICE with repr(align(0))

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' panicked at compiler/rustc_hir_analysis/src/check/check.rs:1625:32: 'called `Option::unwrap()` on a `None` value'', 'thread 'rustc' panicked at compiler/rustc_hir_analysis/src/check/check.rs:1625:32: 'called `Option::unwrap()` on a `None` value''
File: /tmp/im/a.rs
-->

auto-reduced (treereduce-rust):
````rust
#[repr(align(0))]
enum Foo {}

````

original:
````rust
#[repr(align(0))]
enum Foo {}

pub fn main() {}

````

Version information
````
rustc 1.90.0-nightly (733b47ea4 2025-07-05)
binary: rustc
commit-hash: 733b47ea4b1b86216f14ef56e49440c33933f230
commit-date: 2025-07-05
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.7

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/733b47ea4b1b86216f14ef56e49440c33933f230/compiler/rustc_hir_analysis/src/check/check.rs#L1619-L1631

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0589]: invalid `repr(align)` attribute: not a power of two
 --> /tmp/icemaker_global_tempdir.bwtdMDbGmL3S/rustc_testrunner_tmpdir_reporting.jBLUdmulrpMj/mvce.rs:1:14
  |
1 | #[repr(align(0))]
  |              ^

error[E0601]: `main` function not found in crate `mvce`
 --> /tmp/icemaker_global_tempdir.bwtdMDbGmL3S/rustc_testrunner_tmpdir_reporting.jBLUdmulrpMj/mvce.rs:2:12
  |
2 | enum Foo {}
  |            ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.bwtdMDbGmL3S/rustc_testrunner_tmpdir_reporting.jBLUdmulrpMj/mvce.rs`


thread 'rustc' panicked at compiler/rustc_hir_analysis/src/check/check.rs:1625:32:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0:     0x7d4d4c536883 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::ha2e7e3a01df69042
   1:     0x7d4d4cc0273f - core::fmt::write::h7cb8f63788cd01d2
   2:     0x7d4d4c52c4c3 - std::io::Write::write_fmt::h5ebaf381f8338068
   3:     0x7d4d4c5366e2 - std::sys::backtrace::BacktraceLock::print::h3ec4d7883eb25e61
   4:     0x7d4d4c53a17a - std::panicking::default_hook::{{closure}}::h29548987efd832cb
   5:     0x7d4d4c539cff - std::panicking::default_hook::ha25170a15c643514
   6:     0x7d4d4b6056a3 - std[d534d36ffd607ff3]::panicking::update_hook::<alloc[756c202a2770bd7e]::boxed::Box<rustc_driver_impl[4ab48184b7847074]::install_ice_hook::{closure#1}>>::{closure#0}
   7:     0x7d4d4c53a9cb - std::panicking::rust_panic_with_hook::h0d81afcd829aa24b
   8:     0x7d4d4c53a696 - std::panicking::begin_panic_handler::{{closure}}::hc84a33f1202346cf
   9:     0x7d4d4c536d49 - std::sys::backtrace::__rust_end_short_backtrace::h373067a14f6c59aa
  10:     0x7d4d4c53a3ad - __rustc[beb0385846a06d21]::rust_begin_unwind
  11:     0x7d4d48ca8530 - core::panicking::panic_fmt::ha33fa2ae772efba9
  12:     0x7d4d4902e21c - core::panicking::panic::hd2a569f4f37f7e10
  13:     0x7d4d48caa0f9 - core::option::unwrap_failed::ha14784b3b8a80e23
  14:     0x7d4d4d56ce26 - rustc_hir_analysis[a54d78a35e7f2910]::check::check::check_item_type
  15:     0x7d4d4d557777 - rustc_hir_analysis[a54d78a35e7f2910]::check::wfcheck::check_well_formed
  16:     0x7d4d4d557759 - rustc_query_impl[65f50324650e1fa3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[65f50324650e1fa3]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 1usize]>>
  17:     0x7d4d4d556f8d - rustc_query_system[2bb68dc6eee57125]::query::plumbing::try_execute_query::<rustc_query_impl[65f50324650e1fa3]::DynamicConfig<rustc_data_structures[64cba4f9a751aff]::vec_cache::VecCache<rustc_span[439cc0b179e5ffc0]::def_id::LocalDefId, rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[2bb68dc6eee57125]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[65f50324650e1fa3]::plumbing::QueryCtxt, false>
  18:     0x7d4d4d556ac2 - rustc_query_impl[65f50324650e1fa3]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  19:     0x7d4d4d552cc2 - rustc_hir_analysis[a54d78a35e7f2910]::check::wfcheck::check_type_wf
  20:     0x7d4d4d552bdb - rustc_query_impl[65f50324650e1fa3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[65f50324650e1fa3]::query_impl::check_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 1usize]>>
  21:     0x7d4d4dc874fc - rustc_query_system[2bb68dc6eee57125]::query::plumbing::try_execute_query::<rustc_query_impl[65f50324650e1fa3]::DynamicConfig<rustc_query_system[2bb68dc6eee57125]::query::caches::SingleCache<rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[65f50324650e1fa3]::plumbing::QueryCtxt, false>
  22:     0x7d4d4dc872f6 - rustc_query_impl[65f50324650e1fa3]::query_impl::check_type_wf::get_query_non_incr::__rust_end_short_backtrace
  23:     0x7d4d4d40d51c - rustc_hir_analysis[a54d78a35e7f2910]::check_crate
  24:     0x7d4d4d600c23 - rustc_interface[331c37c46c9c8d25]::passes::analysis
  25:     0x7d4d4d60082b - rustc_query_impl[65f50324650e1fa3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[65f50324650e1fa3]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 0usize]>>
  26:     0x7d4d4dc88c7c - rustc_query_system[2bb68dc6eee57125]::query::plumbing::try_execute_query::<rustc_query_impl[65f50324650e1fa3]::DynamicConfig<rustc_query_system[2bb68dc6eee57125]::query::caches::SingleCache<rustc_middle[1c3bc8d0cdd03ead]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[65f50324650e1fa3]::plumbing::QueryCtxt, false>
  27:     0x7d4d4dc88846 - rustc_query_impl[65f50324650e1fa3]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  28:     0x7d4d4de8a9d6 - rustc_interface[331c37c46c9c8d25]::passes::create_and_enter_global_ctxt::<core[a1c5ac6c25c163e6]::option::Option<rustc_interface[331c37c46c9c8d25]::queries::Linker>, rustc_driver_impl[4ab48184b7847074]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}
  29:     0x7d4d4de58c03 - rustc_interface[331c37c46c9c8d25]::interface::run_compiler::<(), rustc_driver_impl[4ab48184b7847074]::run_compiler::{closure#0}>::{closure#1}
  30:     0x7d4d4ddf19f8 - std[d534d36ffd607ff3]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[331c37c46c9c8d25]::util::run_in_thread_with_globals<rustc_interface[331c37c46c9c8d25]::util::run_in_thread_pool_with_globals<rustc_interface[331c37c46c9c8d25]::interface::run_compiler<(), rustc_driver_impl[4ab48184b7847074]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  31:     0x7d4d4ddf16d6 - <<std[d534d36ffd607ff3]::thread::Builder>::spawn_unchecked_<rustc_interface[331c37c46c9c8d25]::util::run_in_thread_with_globals<rustc_interface[331c37c46c9c8d25]::util::run_in_thread_pool_with_globals<rustc_interface[331c37c46c9c8d25]::interface::run_compiler<(), rustc_driver_impl[4ab48184b7847074]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[a1c5ac6c25c163e6]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  32:     0x7d4d4ddefd3d - std::sys::pal::unix::thread::Thread::new::thread_start::h1a22ded422ce395b
  33:     0x7d4d478a57eb - <unknown>
  34:     0x7d4d4792918c - <unknown>
  35:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.90.0-nightly (733b47ea4 2025-07-05) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [check_well_formed] checking that `Foo` is well-formed
#1 [check_type_wf] checking that types are well-formed
#2 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 2 previous errors

Some errors have detailed explanations: E0589, E0601.
For more information about an error, try `rustc --explain E0589`.

```

</p>
</details>

<!--
query stack:
#0 [check_well_formed] checking that `Foo` is well-formed
#1 [check_type_wf] checking that types are well-formed
#2 [analysis] running analysis passes on this crate

-->



<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/tests/ui/attributes/malformed-reprs.rs
+++ b/tests/ui/attributes/malformed-reprs.rs
@@ -1,0 +1,14 @@
+// Tests a few different invalid repr attributes
+
+// This is a regression test for https://github.com/rust-lang/rust/issues/143522
+#![repr]
+//~^ ERROR malformed `repr` attribute input [E0539]
+//~| ERROR `repr` attribute cannot be used at crate level
+
+// This is a regression test for https://github.com/rust-lang/rust/issues/143479
+#[repr(align(0))]
+//~^ ERROR invalid `repr(align)` attribute: not a power of two
+//~| ERROR unsupported representation for zero-variant enum [E0084]
+enum Foo {}
+
+fn main() {}

--- a/tests/ui/attributes/malformed-reprs.stderr
+++ b/tests/ui/attributes/malformed-reprs.stderr
@@ -1,0 +1,43 @@
+error[E0539]: malformed `repr` attribute input
+  --> $DIR/malformed-reprs.rs:4:1
+   |
+LL | #![repr]
+   | ^^^^^^^^
+   | |
+   | expected this to be a list
+   | help: must be of the form: `#[repr(C | Rust | align(...) | packed(...) | <integer type> | transparent)]`
+
+error[E0589]: invalid `repr(align)` attribute: not a power of two
+  --> $DIR/malformed-reprs.rs:9:14
+   |
+LL | #[repr(align(0))]
+   |              ^
+
+error: `repr` attribute cannot be used at crate level
+  --> $DIR/malformed-reprs.rs:4:1
+   |
+LL | #![repr]
+   | ^^^^^^^^
+...
+LL | enum Foo {}
+   |      --- the inner attribute doesn't annotate this enum
+   |
+help: perhaps you meant to use an outer attribute
+   |
+LL - #![repr]
+LL + #[repr]
+   |
+
+error[E0084]: unsupported representation for zero-variant enum
+  --> $DIR/malformed-reprs.rs:9:1
+   |
+LL | #[repr(align(0))]
+   | ^^^^^^^^^^^^^^^^^
+...
+LL | enum Foo {}
+   | -------- zero-variant enum
+
+error: aborting due to 4 previous errors
+
+Some errors have detailed explanations: E0084, E0539, E0589.
+For more information about an error, try `rustc --explain E0084`.

